src/util/psbt/map/output.rs

```diff
@@ -117,6 +118,45 @@ impl TapTree {
     pub fn into_inner(self) -> TaprootBuilder {
         self.0
     }
+
+    /// Returns iterator for a taproot script tree, operating in DFS order over leaf depth and
+    /// leaf script pairs.
+    pub fn iter(&self) -> TapTreeIter {
+        self.into_iter()
```

Comment:
Are we happy that this complies with Rust convention that `iter` should return an iterator that iterates over `&T`? I spent some time reading the docs for `IntoIterator` and I can't work out if the tuple `(u8, &'tree Script)` is in line with convention or not. Should it by `(&u8, &'tree Script)`?

Reply:
It can't be u8, since u8 is not owned by the `TapTree` structure and is just retured from `len()` method which instantiate the value. You technically can't return a reference to it.

Reply:
My bad, thanks.